awscrt/s3.py

```diff
@@ -180,6 +180,11 @@ class S3Client(NativeResource):
             Gigabits per second (Gbps) that we are trying to reach.
             You can also use `get_recommended_throughput_target_gbps()` to get recommended value for your system.
             10.0 Gbps by default (may change in future)
+
+        memory_limit (Optional[int]): Memory limit, in bytes, of how much memory
+            client can use for buffering data for requests.
+            Default values scale with target throughput and are currently
+            between 2GBs and 8GBs (might change in future.)
```

Comment:
Should we have a minimum value? Like anything less than 2GB will be bumped up to 2GB.

Reply:
I though about it before, but im on a fence on that. Mostly because i didnt want to complicate logic. Right now its fairly straight forward, and we figure out number if you dont provide one and we respect what you provided otherwise (with some sanity checking). Lets backlog it for now?